UTIL: rename & cleanup sysdb_error_to_errno

thalman opened the pull request:
* UTIL: move and rename sysdb_error_to_errno to utils
* UTIL: Use new ldb_error_to_errno function in code

mzidek-rh commented on src/util/util_errors.h:
```diff
@@ -178,4 +178,10 @@ enum sssd_errors {
+errno_t ldb_error_to_errno(int ldberr);
+
+/* define old name for backward compatibility */
+#define sysdb_error_to_errno(ldberr) ldb_error_to_errno(ldberr)
```
I would prefer to remove the old name. So for the short term backward compatibility it does not matter much where it is IMO. Do you agree?

thalman replied:
This will make the diff really huge. Db module is full of this call.

mzidek-rh replied:
I would prefer to drop the name, but I do not have strong opinion on it and I see it can have some downsides when backporting patches to older branches. To move this PR forward I would do the following:
- Move the #define alias to the sysdb.h where the original function was declared
- create an issue to remove it later (maybe before some downstream rebase would be a good time)
- add FIXME comment before the alias is #defined with link to the issue.

We can then discuss in the new issue if and when it makes sense to drop the name.

mzidek-rh commented on src/util/util_errors.c:
```diff
@@ -146,3 +147,27 @@ const char *sss_strerror(errno_t error)
 }
 
+/* TODO: make a more complete and precise mapping */
+errno_t ldb_error_to_errno(int ldberr)
```
True, I was not thinking about that. I suck at creating names for functions, but I like the sss_ldb2errno to keep the name short. I think it is obvious that by ldb we mean ldb_error if we convert it to errno.

thalman replied:
I like sss_ldb_..., it will have the same prefix as the other function in util_errors.[ch]

mzidek-rh replied:
Ok, jhrozek also suggested it above, so let us go with sss_lldb_*

thalman commented:
I will take a look at it and extend PR

jhrozek commented:
> On 3 Oct 2018, at 13:23, mzidek-rh <notifications@github.com> wrote:
> 
> Hi, the patches look good and in general I have no problem accepting them (when CI finishes). But I would like to see follow up pull request that these two things:
> 
> 	• create one patch for every file that uses sysdb_error_to_errno to use the new function (or alternatively one big patch for all files)

I’m torn on this. On one hand, this would be nice to have a clean code, but I’m a bit afraid that it would make backporting patches from sysdb code to 1.16 harder later and create conflicts. But since the sysdb code is relatively stable, it might be good to do this actually.

> 	• as last patch, remove the sysdb_error_to_errno alias (the sysdb_* prefix is just obfuscating IMO and we should not use in new code)
> Or you can do the above in this PR as well (up to you). If you do not plan to do the follow up PR soon then please make a ticket so that we track it for later.
> 
> Let me know how you prefer to proceed.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mzidek-rh commented:
> I’m torn on this. On one hand, this would be nice to have a clean code, but I’m a bit afraid that it would make backporting patches from sysdb code to 1.16 harder later and create conflicts. But since the sysdb code is relatively stable, it might be good to do this actually.

True, there could be conflicts because of this, but resolving them should be quite easy, just replaceing the function name in the diff/patch to have the old name should do the trick. Still a bit annoying, but I think it is worth getting rid of the old name.

thalman commented:
@mzidek-rh, @jhrozek PR updated, I moved #define to sysdb.h. I also did the cleanup.

Please take a look at cleanup changes. Since I use sss_ldb_error_to_errno in some places, modified functions can eventually return more error values.

For example old code:
https://github.com/SSSD/sssd/blob/dfa7bf1133f002a9fbbd3495a70909913db25b16/src/util/secrets/secrets.c#L419-L431

After my changes it can theoretically return also EFAULT.

jhrozek commented:
On Mon, Oct 08, 2018 at 01:56:37AM -0700, Tomáš Halman wrote:
> @mzidek-rh, @jhrozek PR updated, I moved #define to sysdb.h. I also did the cleanup.
> 
> Please take a look at cleanup changes. Since I use sss_ldb_error_to_errno in some places, modified functions can eventually return more error values.

This looks mostly good. But one thing that I don't like very much is
that krb5_child now needs to link with ldb libs.

Do you think having util/util_ldb_errors.c to avoid this extra
dependency is too excessive?

> 
> For example old code:
> https://github.com/SSSD/sssd/blob/dfa7bf1133f002a9fbbd3495a70909913db25b16/src/util/secrets/secrets.c#L419-L431
> 
> After my changes it can theoretically return also EFAULT. 

Right. But I think this is fine. I don't think any of the callers cared
about any special error code other than EOK and maybe ENOENT.

jhrozek commented:
retest this please

jhrozek commented:
Aah, now I know what you mean. Yes, I agree that the numerical code is preferable over the dependency.

thalman commented:
PR updated

jhrozek commented:
Hey @mzidek-rh if you don't have any more comments, I'll just push this patch as it is.

jhrozek commented:
I'm going to assume the silence means "I'm OK" and push the patch

jhrozek commented:
* master: f62f3b290cb5393bf9c4a9c2c424fd5c5185fe44
